OSDOCS-19870: adds main console docs RHCL (#112113)
openshift/openshift-docs: modules/con-rhcl-api-management-console-pages.adoc

+
Use the *My API Keys* page to complete the following tasks:
+
* Browse your requested API keys
* Retrieve your approved credentials
* View authentication requirements and service tiers
* Request access to APIs

API owner:: You are responsible for specific services. Control how your APIs are presented and accessed through the *API Products* and *API Key Approvals* pages. You can create new API products, view and edit existing products, configure approval modes, and access detailed product information.
+
The *API Products* page is located in the main *{prodname}* section and lists all of the `APIProduct` resources in the cluster or current namespace. Use the *API Products* page to complete the following tasks:
+
* Define how the API is displayed in the catalog through `APIProduct` resource metadata
* Choose between automatic and manual approval for access requests
* Approve or reject requests
* Set documentation links so that API consumers can self-serve

The *API Key Approvals* page is located in the *{prodname} API Catalog* section of the web console. Use the *API Key Approvals* page to complete the following tasks:
+

> * Review pending `APIKey` requests

Reviewer: I might be looking at some outdated render, but in https://112113--ocpdocs-pr.netlify.app/rhcl/latest/develop/rhcl-api-management this list is not correctly formatted

<img width="617" height="203" alt="Image" src="https://github.com/user-attachments/assets/3fbfd850-636c-4178-ae01-ae2e228a89cf" />

// Module included in the following assemblies:
//
// * develop/rhcl-api-management.adoc

:_mod-docs-content-type: CONCEPT
[id="con-rhcl-api-management-console-pages_{context}"]

> = Understanding API management pages in the web console

Reviewer 2: Since this is a concept, consider removing the gerund from the title. I think it works simply as "API management pages in the web console". If you're not comfortable with that, maybe "Overview of API management pages in the web console"?

[role="_abstract"]
By using the {ocp} web console with {prodname}, different users conduct various API-management tasks, depending on each user's role-based access and requirements.

> The following are examples of how you can interact with the {prodname} API management section of the web console:

Reviewer 2: ```suggestion
The following examples illustrate how you can interact with the {prodname} API management section of the web console:
```

> Everyone can explore available APIs without requesting access to them by browsing the API catalog to discover what services are available. Use the *API Products* page located in the main *{prodname}* section to see all of the `APIProduct` resources in the cluster or current namespace.

Reviewer 2: ```suggestion
Everyone can explore available APIs without requesting access to them by browsing the API catalog to discover what services are available. Use the *API Products* page in the main *{prodname}* section to see all of the `APIProduct` resources in the cluster or current namespace.
```

> API consumers:: Developers who need to integrate with services provided by other teams are API consumers. You can use the *My API Keys* page located in the *{prodname} API Catalog* section to see your API key requests.

Reviewer 2: ```suggestion
API consumers:: Developers who need to integrate with services provided by other teams are API consumers. You can use the *My API Keys* page in the *{prodname} API Catalog* section to see your API key requests.
```

+
Use the *My API Keys* page to complete the following tasks:
+
* Browse your requested API keys
* Retrieve your approved credentials
* View authentication requirements and service tiers
* Request access to APIs

API owner:: You are responsible for specific services. Control how your APIs are presented and accessed through the *API Products* and *API Key Approvals* pages. You can create new API products, view and edit existing products, configure approval modes, and access detailed product information.
+
--
The *API Products* page is located in the main *{prodname}* section and lists all of the `APIProduct` resources in the cluster or current namespace. Use the *API Products* page to complete the following tasks:

* Define how the API is displayed in the catalog through `APIProduct` resource metadata
* Choose between automatic and manual approval for access requests
* Approve or reject requests
* Set documentation links so that API consumers can self-serve

> The *API Key Approvals* page is located in the *{prodname} API Catalog* section of the web console. Use the *API Key Approvals* page to complete the following tasks:

Reviewer 2: ```suggestion
The *API Key Approvals* page is in the *{prodname} API Catalog* section of the web console. Use the *API Key Approvals* page to complete the following tasks:
```

* Review pending `APIKey` requests
* Approve or reject requests
* Add optional comments
* View request details
* Filter requests by status and API product
* Bulk approve or reject requests
* Revoke previously granted access
--

> API administrator:: You give cross-team oversight and governance. Use the *API Key Approvals* page with elevated permissions. The *API Key Approvals* page is located in the *{prodname} API Catalog* section of the web console. Use the *API Key Approvals* page to complete the following tasks:

Reviewer 2: ```suggestion
API administrator:: You give cross-team oversight and governance. You need elevated permissions to use the *API Key Approvals* page. The *API Key Approvals* page is in the *{prodname} API Catalog* section of the web console. Use the *API Key Approvals* page to complete the following tasks:
```
Feel free to ignore my revision if it isn't accurate.

// Module included in the following assemblies:
//
// * develop/rhcl-api-management.adoc

:_mod-docs-content-type: CONCEPT
[id="con-rhcl-api-management-console-pages_{context}"]
= Understanding API management pages in the web console

[role="_abstract"]
By using the {ocp} web console with {prodname}, various API-management tasks are conducted by different users, depending on their access and requirements.

The following are examples of how you can interact with the {prodname} API management section of the web console:

> API browser:: You can explore available APIs without requesting access to them by browsing the API catalog to discover what services are available. Use the *API Products page* located in the main *{prodname}* section to see all of the `APIProduct` resources in the cluster or current namespace.

Reviewer 3: This is a little confusing to me, when i see without requesting access, im thinking RBAC access which they do need i doesnt come accross as APIKeys access. Im thinking to we just leave the examples to the 3 known personas consumer owner and admin

API consumers:: Developers who need to integrate with services provided by other teams are API consumers. You can use the *My API Keys* page located in the *{prodname} API Catalog* section to see your API key requests.
+
Use the *My API Keys* page to complete the following tasks:
+

> * Browse available API products

Reviewer: Browse my requested api keys

Reviewer: Also get access to the credentials for those api keys approved

* View authentication requirements and service tiers

> * Request access to APIs by creating `APIKey` custom resource definitions (CRDs). An `APIKey` CRD represents the actual API access credentials in the consumer's namespace. It includes the `Secret` reference and is created by the controller when an access request is approved.

Reviewer: I would not go into much details. The important point is that My API Keys page allows you to request access to the api key. 

> It includes the `Secret` reference and is created by the controller when an access request is approved.

This is not entirely correct.

> * Retrieve approved credentials

Reviewer 3: You cant browse api products or view authentication requirements you can just see the api key requests they own which can be further clicked into to see more detail but its a different view

API Owner:: You are responsible for specific services. Control how your APIs are presented and accessed through the *API Products* and *API Key Approvals* pages.
+
The *API Products* page is located in the main *{prodname}* section and lists all of the `APIProduct` resources in the cluster or current namespace. Use the *API Products* page to complete the following tasks:
+
* Define how the API is displayed in the catalog through `APIProduct` resource metadata

> * Choose between automatic and manual approval for access requests

Reviewer 3: This is kind of repeating in less detail what your saying below on line 34

Author: not really; this is _what_ you can do, the page list tells you _how_; deliberately separate (easier to spot and update when something changes)

> * Review pending `APIKey` requests

Reviewer: This belongs to API Key Approvals.

However, it is also true that when you go to the details page of one api product, there is a dedicated tab for approving/denying requests. But this is not available in the page with the list of api products

> * Approve or reject requests by creating `APIKeyApproval` CRDs

Reviewer: I would not go into too much detail about _how_ is the approval process happening behind the scenes.

Author: moved CRDs to a reference module

> * Approve or reject requests by creating `APIKeyApproval` CRDs

Reviewer 3: i would also rephrase as yes techincally thr user is creatings crs but not in the yaml sense which this eludes to

* Set documentation links so that API consumers can self-serve
* API owners can create new API products, view and edit existing products, configure approval modes, and access detailed product information through tabs including *Overview*, *Target*, *Definition*, *Policies*, and *API Keys*.
+

> `APIProduct` custom resource definitions (CRDs) wrap an existing `HTTPRoute` custom resource (CR) with the business context needed for consumption, including a human-readable name, documentation links, contact information, and access policies. When an API owner creates an `APIProduct` resource and sets the `publishStatus` field to `Published`, the API becomes discoverable in the catalog.

Reviewer 3: Theres one APIProduct CRD, it would be better to say CR

+
The *API Key Approvals* page is located in the *{prodname} API Catalog* section of the web console. Use the *API Key Approvals* page to complete the following tasks:
+
* Review pending API access requests
* Approve or reject requests

> * Add optional comments

Reviewer 3: This is not a standalone feature its done as part of reject or approve

Author: these aren't features, they are possible user actions

* View request details
* Filter requests by status and API product

> * Bulk approve or reject requests

Reviewer: You can also revoke previously granted access.

+

> An `APIKeyRequest` CRD is a shadow resource in the API owner namespace that enables discovery without exposing API key values. These resources are managed by the {prodname} controller and must not be created or modified manually.

Reviewer: I see this section explaining existing CRDs out of scope here. Maybe have a dedicated section about the existing CRDs for developing APIs with the web console.

+
An `APIKeyApproval` CRD represents an approval or rejection action on an `APIKeyRequest` resource. API owners and administrators create `APIKeyApproval` resources to approve or reject pending requests.

API Administrator:: You give cross-team oversight and governance. Use the *API Key Approvals* page with elevated permissions. The *API Key Approvals* page is located in the *{prodname} API Catalog* section of the web console. Use the *API Key Approvals* page to complete the following tasks:
+
* View and manage all of the `APIProduct` resources across the organization
* Approve or reject any API key request

> * Troubleshoot issues on behalf of API owners

Reviewer 3: This seems a bit of a odd thing to mention as theres no monitoring or analytics they can see that a owner couldn see on their own apiproduct
